recovery-contracts/RecoveryWallet.sol

```diff
@@ -0,0 +1,65 @@
+pragma solidity 0.8.10;
+import "./RecoveryForwarder.sol";
+import "./CloneFactory.sol";
+/**
+ *
+ * RecoveryWallet
+ * ============
+ *
+ * Basic singleSig wallet designed to recover funds.
+ *
+ */
+contract RecoveryWallet is CloneFactory {
+
+  // Public fields
+  address public immutable signer;
+  address public immutable forwarderImplementationAddress;
+
+  constructor (address _signer, address _forwarderImplementationAddress) {
+    signer = _signer;
+    forwarderImplementationAddress = _forwarderImplementationAddress;
+  }
+  /**
+   * Modifier that will execute internal code block only if the sender is an authorized signer on this wallet
+   */
+  modifier onlySigner {
+    require( signer == msg.sender, 'Non-signer in onlySigner method');
+    _;
+  }
+
+  /**
+   * Gets called when a transaction is received with ether and no data
+   */
+  receive() external payable {
+  }
+
+  /**
+   * Creates new forwarder contract controlled by signer in batches
+   */
+function createForwarder(uint8 value) external {
+    require(value > 0 && value < 150 , 'value must be greater than 0 and less than 150');
+    for ( uint8 i = 0; i < value; ++i) {
+    address payable clone = createClone(forwarderImplementationAddress);
+    RecoveryForwarder(clone).init(signer);
+    }
```

Comment:
We can control the number of forwarder created as gasLimit per block varies with respect to evm like chains